react-game/src/components/TrialRow/TrialRow.js

```diff
@@ -0,0 +1,113 @@
+import { useState, useRef, useEffect } from 'react';
+import CheckButton from '../Buttons/CheckButton';
+import WhiteCircle from '../ResultCircles/WhiteCircle';
+import BlueCircle from '../ResultCircles/BlueCircle';
+import EnteredNumber from './EnteredNumber';
+import classes from './Trial.module.css';
+
+
+function TrialRow({ rowIndex, selectedIndex, setSelectedIndex, secretCode, setGameStatus, clearScreen, setClearScreen }) {
+
+    //STATES & HOOKS------------------------------------------------------------------
+    const [correctNumberWithIndex, setcorrectNumberWithIndex] = useState(0);
+    const [correctNumberOnly, setcorrectNumberOnly] = useState(0);
+    const [rowEnabled, SetRowEnabled] = useState(false);
+    const enteredNumberRef = useRef([]); //to get the entered number from the number input 
+
+    useEffect(() => { //check to enable the row or not
+        if (rowIndex === selectedIndex) {
+            SetRowEnabled(true);
+            if (rowIndex === 1) {
+                console.log("The secret code is: " + secretCode);
+            }
+        }
+    }, [rowIndex, selectedIndex]);
+
+    useEffect(() => { //check to clear the row or not
+        if (clearScreen === true) {
+            setcorrectNumberWithIndex(0);
+            setcorrectNumberOnly(0);
+            setClearScreen(false);
+        }
+        if (rowIndex !== 1) {
+            SetRowEnabled(false);
+        }
+    }, [clearScreen]);
+
+
+
+    //FUNCTIONS-----------------------------------------------------------------
+    const checkButtonClickHandler = () => {
+
+        //get the entered number from the reference array
+        const enteredNumberArray = [];
+        for (let i = 0; i < 4; i++) {
+            enteredNumberArray[i] = enteredNumberRef.current[i].value;
+        }
+
+        var secretCodeArray = secretCode.toString().split(""); //convert the secretCode from number to array of digits
+        let correctNumbersIndexCount = 0;
```

Comment:
please use const instead of let

Reply:
This change gives an error, since I am reassigning (correctNumbersIndexCount) by incrementing it